### ISSUE TITLE
Realm notifications don't automatically sorted with new added object

### ISSUE THREAD

**Issue by peihsendoyle** [realm/realm-cocoa#5069]
I have a table view and results get from Realm query with a sorted criteria as "createdAt" ascending. I integrate the results array with Notification so that it automatically update the tableView.

The problem here is when I add new object with the "createdAt" property as Date() (it means today time) to Realm, the notification get fired but it update the new object to the last row of tableView (although I sort the results with: realm.objects().sorted(byKeyPath: "createdAt) ). 
I want it to the first row (first index) of tableView.

Maybe I don't know exactly how Realm notifications work, anyone can help me clear this case? 

P/s: Because I'm on board now to go home, so I'm sorry that I cannot provide the code I used on tableView. I will provide it when I come home, thanks all.

**Comment by nirinchev** [comment]
If `createdAt` behaves as expected, then I would assume the newly created object will have the largest `createdAt` value (because it was created last). This means that an ascending (smallest to largest) ordering will correctly put it as the last object in the collection. Perhaps you meant to order them in descending order?

**Comment by peihsendoyle** [comment]
@nirinchev thanks for your help.

When I put the order in `realm.objects().sorted(byKeyPath: "created_at", ascending: true)` ( this is ascending order, right?). The collection is in the right order (from nearest time to furthest time as below picture). But when the new notification come in (I add new object to Realm), it is put at the bottom of the collection.
<img width="314" alt="screen shot 2017-06-26 at 8 18 11 am" src="https://user-images.githubusercontent.com/7514128/27521528-1cfa4072-5a48-11e7-9c24-bdfd6490e56f.png">

Vice versa, when I put `realm.objects().sorted(byKeyPath: "created_at", ascending: false)` (descending), the collection is reversed by time as below second picture (i don't want it show from farthest to nearest time). But when the new notification come, it is put at the first row (I want it).
<img width="320" alt="screen shot 2017-06-26 at 8 27 32 am" src="https://user-images.githubusercontent.com/7514128/27521613-51682256-5a49-11e7-9758-45f1819ba745.png">

So could you tell me what was wrong here, I want the collection show from nearest time to furthest time, and the new object put at the first order. Thanks.

**Comment by nirinchev** [comment]
It's not immediately obvious from your screenshots if the values are in the past or in the future (e.g. 2 months ago or 2 months from now). I would assume in the past, but then the ordering doesn't make sense - for example if I have 3 dates:
1. June 5, 2017
2. June 15, 2017
3. Today - June 26, 2017

An ascending ordering would arrange them like:
1, 2, 3 - the oldest date will be first, effectively rendering a UITableView that should look like:
```
3 weeks
2 weeks
today
```

But that is the opposite of what you report :/

Are you by any chance reversing the collection or the UITableView or am I misunderstanding the meaning of `created_at`?

**Comment by peihsendoyle** [comment]
@nirinchev thanks very much, I think I've cleared out what was wrong here. 

The `NotificationModel` which is subclassed of Object (Realm) has a property `created_at` of `Date` (swift 3). After received results from APIs I mapped it with ObjectMapper (uses `DateTransform()`). Because of the transform, the date of new object (we add on the client, not server) is not the same type as the results from server (which are transformed).

I've changed the sorted keypath to `object_id` (descending) and everything worked fine. But I assume that if I could use the date sorting in the future. If there is anyway to map the `Date` property from JSON with no transformation?

Thanks for you help again.
